jdbc/src/main/java/nextstep/jdbc/JdbcTemplate.java

```diff
@@ -14,4 +20,49 @@ public class JdbcTemplate {
     public JdbcTemplate(final DataSource dataSource) {
         this.dataSource = dataSource;
     }
+
+    private Connection getConnection() throws SQLException {
+        return dataSource.getConnection();
+    }
+
+    public int insert(final String sql, final Object... params) {
+        try (
+                final var connection = getConnection();
+                final var pstmt = connection.prepareStatement(sql);
+        ) {
+            log.debug("query : {}", sql);
+            setParams(pstmt, List.of(params));
+
+            return pstmt.executeUpdate();
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    public <T> T select(final String sql, final Function<ResultSet, T> mapper, final Object... params) {
+        try (
+                final var connection = getConnection();
+                final var pstmt = connection.prepareStatement(sql);
+        ) {
+            log.debug("query : {}", sql);
+            setParams(pstmt, List.of(params));
+            final var resultSet = pstmt.executeQuery();
+
+            return mapper.apply(resultSet);
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void setParams(final PreparedStatement pstmt, final List<Object> params) throws SQLException {
+        if (Objects.isNull(params)) {
+            return;
+        }
+
+        for (int i = 0; i < params.size(); i++) {
+            pstmt.setObject(i + 1, params.get(i));
+        }
```

Comment:
저랑 살짝 다르게 구현하셔서 제 방법도 공유할게요😁
저는 for문 i 숫자 세는게 귀찮아서 foreach문을 써봤습니다!

```suggestion
        int index = 1;
        for (Object arg : args) {
            pstmt.setObject(index++, arg);
        }
```

Reply:
오 아서는 이렇게 구현하셨군요! 👍
저는 개인적으로 반복문 내부에서만 사용되는 변수를 활용하는 방법이 더 낫지 않을까하는 생각이 들어 유지하도록 하겠습니다!